main.tex

```diff
@@ -578,105 +732,156 @@ \subsubsection{Malicious Coordinator}
+
+A malicious coordinator could also delay the processing of requests in order to learn more through timing and ordering leaks. In the worst case, the coordinator can attempt to linearize all requests by delaying individual to recover the full set of labelled edges. This is possible when $k=1$ and users have minimal dependencies between their requests and tolerate arbitrary timeouts but issue requests in a timely manner.
+
+Similarly the coordinator may delay information such as the set of ownership proofs or the final unsigned transaction. In the case of the latter, this can be used to learn about links between inputs. This is because a signature can only be made after the details of the transaction are known. If the unsigned was only known to one user but multiple inputs have provided signatures, it follows that those inputs are owned by the same user.
+
+Since the coordinator must trusted with regards to denial of service a more practical variant of this attack would involve more subtle delays followed by sabotaging multiple successive rounds during the signing phase in order to learn of correlations between registrations while maintaining deniability.
+
+More generally denial of service can amplify attacks on unlinkability, as it can be used to perform intersection attacks. The coordinator always learns the requested inputs and outputs, even if a round fails, and is able to partition users. A malicious participant will also learn all the input and output registrations if they wait until the signing phase to defect\footnote{If \texttt{SIGHASH\_ANYONECANPAY} is set in the signature flags the full set of inputs could be kept known only to the coordinator until all signatures have been provided.}.
+
+\subsection{Theft prevention}
+
+Since the output of the protocol is a single Bitcoin transaction theft is prevented by only signing the transaction if the outputs are as expected, so the protocol inherits theft resistance directly from Bitcoin's security model.
+
+A malicious user could claim more funds than she registered if and only if she is able to forge the coordinator's MAC on some credential. This is certainly not possible unless credentials can be forged under chosen message attacks, but even if that were the case this would only result in denial of service, as the honest users would refuse to sign such a transaction.
+
+\subsection{Security Proofs}
+The above outlined security and privacy guarantees directly follow from the applied KVAC scheme of~\cite{chase2019signal}. The unlinkability of WabiSabi credentials follow from the unlinkability of the underlying keyed-verification anonymous
+credential scheme. The theft prevention of the WabiSabi protocol follows from the unforgeability of the KVAC scheme. We refer the to~\cite{chase2019signal} for formal security arguments.
```

Comment:
I think the theft prevention comes from the fact that users will not sign a transaction that takes their money and from the fact that the bitcoin network will reject an inflationary transaction. I don't think the KVAC takes part in theft prevention.

Reply:
wat, how did that get in there, you're right this is a serious mistake

Reply:
I think it's supposed to say `double spending prevention`...

Reply:
nope, that's also incorrect.

the only things the credentials ensure is that no unauthorized output registrations are attempted.

if the coordinator is honest, which means only accepts authorized registrations and does not reject any of them, all honest clients should sign the resulting transaction.

all this ensures is that if the coordinator is honest it can still prevent unauthorized registrations, which would either make the transaction invalid or force an honest participant not to sign, and it's only the signature that has anything to do with theft or double spending prevention.

Reply:
put another way, if an adversary is able to prove knowledge of a forged MAC, they would be able to register outputs without registering an input or receiving a credential as part of a payment, and an honest coordinator would not be able to output a transaction that all honest users would sign, allowing the adversary to deny service to honest participants indefinitely (i.e. total failure of the protocol).

Reply:
```suggestion
credential scheme. The denial of service guarantees of the WabiSabi protocol follow from the unforgeability of the KVAC scheme. We refer the to~\cite{chase2019signal} for formal security arguments.
```